### PR TITLE
AppID concept doc

### PR DESCRIPTION
**Pull request by ptimmons**
Includes update to sidebars.

### CODE REVIEW

File: docs/concepts_appid.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            exit
+            share-service-routes  false
+        exit
+    exit
+exit
+```
+
+In this configuration, the `address` is configured as `community.128technology.com`. This will be resolved once the configuration is committed, and the 128T will resolve that hostname and install FIB entries. The 128T will periodically re-resolve those hostnames to ensure that the FIB entries are kept up-to-date.
+
+> Note: the names used in the `address` field must be DNS-resolvable names. You cannot use wildcards such as `*.128technology.com`.
```
Comment (MichaelBaj) [inline docs/concepts_appid.md:48]: unfortunately they don't render well in Typora, but Docusaurus uses something called [admonitions/callouts](https://v2.docusaurus.io/docs/markdown-features/#calloutsadmonitions).  I have been using them throughout the docs site.  In this case you would use `:::note` followed by text and a trailing `:::`.

File: docs/concepts_appid.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+2. Unlike AppID based on TLS, there is no "chicken and egg issue;" i.e., no prerequisite to have the destination be reachable in order to learn the destination's name.
+
+**Disadvangates**:
+
+This technique relies on your 128T getting the same answers from DNS that your clients get. Take for example a massive SaaS provider such as `www.salesforce.com`. Their web site resolves to hundreds or thousands of IP addresses distributed across the globe. When the 128T resolves that hostname, it will get a handful of answers; if a client does the same DNS query and gets a different set of answers, then the traffic the client sends will not match the FIB entries that the 128T has created.
+
+*Key to Success*: Use DNS-based services when your destination uses a small set of nonvolatile IP addresses. This technique is perfect for small, hosted services. The effectiveness of using DNS-based services diminishes rapidly when referencing names that resolve to dozens of address or more.
```
Comment (MichaelBaj) [inline docs/concepts_appid.md:59]: Following the above comment, I have been treating the Key To Success (I like this) as a callout using the syntax ```:::tip Key to success`.  You can define your own heading using the styling of a different admonition.

File: docs/concepts_appid.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+### AppID based on TLS
```
Comment (MichaelBaj) [inline docs/concepts_appid.md:61]: Note: The new APP ID functionality will be changing this behavior.  I believe you have been involved in the discussions.  For the purposes of the current version of software, this documentation is correct.

File: docs/concepts_appid.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+            application-identification
+                mode  module
+            exit
+        exit
+    exit
+exit
+```
+
+Next, scripts are placed on the router's filesystem at `/etc/128technology/application-modules/`. These scripts must produce JSON output stored at `/var/run/128technology/application-modules/`, which in turn will be consumed by the 128T router and installed as FIB entries.
```
Comment (MichaelBaj) [inline docs/concepts_appid.md:181]: you may want to add that while not as dynamic as using a script, you can also just add a static JSON file to the `application-modules` directory to define service addresses.

File: docs/concepts_appid.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        "ip-prefix": "3.22.11.0/24"
+      },
+      ...
+    ]
+  },
+  "module-name": "zoom"
+}
+```
+
+> Note: in the case of this module, the script that generates the JSON is `/etc/128technology/application-modules/zoom.py`, the output is stored as `/var/run/128technology/application-modules/zoom.json`
```
Comment (MichaelBaj) [inline docs/concepts_appid.md:232]: would you consider adding the contents of `zoom.py` as an appendix to this guide?

File: docs/concepts_appid.md (hunk 2 above)
Comment (ptimmons) [reply docs/concepts_appid.md:59]: Nice, will do.

File: docs/concepts_appid.md (hunk 3 above)
Comment (ptimmons) [reply docs/concepts_appid.md:61]: Yeah, I'm just documenting what's there now with the understanding that it'll need to be rewritten for new behavior.

File: docs/concepts_appid.md (hunk 4 above)
Comment (ptimmons) [reply docs/concepts_appid.md:181]: office365 includes a python executable too.

Comment (ptimmons) [reply docs/concepts_appid.md:181]: office365 still relies on a python module too.

**Pull request merged**